recipes/vtu11/all/conanfile.py

```diff
@@ -0,0 +1,80 @@
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd, valid_min_cppstd
+from conan.tools.files import export_conandata_patches, get, copy, rm
+from conan.tools.layout import basic_layout
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class PackageConan(ConanFile):
+    name = "vtu11"
+    description = "Vtu11 is a small C++ header-only library to write unstructured grids using the vtu file format"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/phmkopp/vtu11"
+    topics = ("vtu", "c++11")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "with_zlib": [True, False],
+    }
+    default_options = {
+        "with_zlib": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 11
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.13", transitive_headers=True)
+
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def _patch_sources(self):
+        if valid_min_cppstd(self, 17):
+            rm(self, "filesystem.hpp", os.path.join(self.source_folder, "vtu11", "inc"))
```

Comment:
This should always be removed so the version from CCI can be required https://github.com/conan-io/conan-center-index/blob/9fe022bf61dc19e010922812163e1ad05c2989ea/recipes/ghc-filesystem/all/conandata.yml#L2

```suggestion
        rm(self, "filesystem.hpp", os.path.join(self.source_folder, "vtu11", "inc"))
```

Reply:
I also noticed this can be patched to work off of Conan settings 

https://github.com/phmkopp/vtu11/blob/a5e82006021036ea8f14025a30d64487415c5a6d/vtu11/inc/filesystem.hpp#L29

Instead of removing the file completely

Reply:
https://github.com/phmkopp/vtu11/blob/a5e82006021036ea8f14025a30d64487415c5a6d/vtu11/inc/alias.hpp#L43

Yeah thats definitely how it works

Reply:
@prince-chrismc I'm indeed gonna fix this with a patch. This fails on some Windows systems because MSVC needs this to actually report the right value when using the __cplusplus pre-definition macro. See here for details: https://learn.microsoft.com/en-us/cpp/build/reference/zc-cplusplus